lib/Dialect/TTNN/Transforms/Workarounds/Decomposition/ReduceOpsRewritePattern.cpp

```diff
@@ -0,0 +1,73 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "ttmlir/Dialect/TTNN/Transforms/Workarounds/Decomposition/ReduceOpsRewritePattern.h"
+
+#include <algorithm>
+#include <unordered_set>
+
+namespace mlir::tt::ttnn::workarounds::decomposition {
+
+llvm::SmallVector<int64_t>
+getReduceDims(const std::optional<mlir::ArrayAttr> &dimArg) {
+  llvm::SmallVector<int64_t> reduceDims;
+  if (!dimArg.has_value()) {
+    return reduceDims;
+  }
+
+  for (const mlir::Attribute &reduceDim : dimArg.value()) {
+    reduceDims.push_back(mlir::cast<mlir::IntegerAttr>(reduceDim).getInt());
+  }
+
+  return reduceDims;
+}
+
+std::vector<int64_t>
+calculateNewReduceShape(const RankedTensorType &inputType,
+                        const std::optional<mlir::ArrayAttr> &dimArg) {
+  std::vector<int64_t> outputShapeVec = inputType.getShape().vec();
```

Comment:
I would use `llvm::SmallVector<int64_t> outputShapeVec(inputType.getShape())` here, as `vec()` still constructs a new `std::vector`, so i don't see any advantage of using it over `llvm::SmallVector`.

Reply:
I wanted to use `llvm::SmallVector` but wasn't sure if I will be able to use `std::fill`, now I see I can, thanks!